### PR TITLE
doc for exponential search

### LINKED ISSUE #173: Content: Exponential Search
## 📚 Documentation

Add documentation page for exponential search.

What is exponential search?
An explanation for exponential search
Algorithmic Steps
Code (Use C++/Java and Python; At least two languages should be present and pick up code from NeoAlgo with apt credits to author)
Sample I/O
Time & Space Complexity
This article will be between 500-700 words in length

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation page explaining Exponential Search Algorithm.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #173

### CODE REVIEW

File: docs/searching-algorithms/exponential-search.mdx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    - index = 2*2 = 4, 11 <= 11 is true
+
+    | 11 | 13 | 15 | 17 | 19 |
+    |----|----|----|----|----|
+
+    - index = 2*4 = 8, 19 <= 11 is false
+    So we have found our range that is from 8/2 = 4 to 8
+
+    | 11 | 13 | 15 | 17 | 19 |
+    |----|----|----|----|----|
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:39]: Keep this whole section in an `Explanation` section

File: docs/searching-algorithms/exponential-search.mdx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+## Algorithm
+
+These are the main steps for this algorithm:
+- Start with first element and check if its the one that needs to be searched.
+- If not, start with the 2nd element (index = 1) and if the item is larger, double the index.
+- Keep doubling till the item to be searched is lower than or equal to the element at the index, that is, till `item <= arr[index]`.
+- Now the range from that `index / 2` to `index` is where the item exists if its present in the array. we start with half of `index` as we have already compared and found that the item is greater than half of `index`.
+- Apply binary search on the above range to get the result
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:48]: Put a full stop at the end of each sentence

File: docs/searching-algorithms/exponential-search.mdx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      print(exponential_search(sorted(search_list), 0, len(search_list)-1, search_value))
+  main()
+  ```
+</TabItem>
+</Tabs>
+
+
+## Sample I/O
+
+### Sample Input
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:216]: Keep Heading-4

File: docs/searching-algorithms/exponential-search.mdx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+```bash
+Enter the size of array
+8
+Enter the sorted array
+5 12 25 38 52 76 110 155
+Enter the element to search
+52
+```
+
+### Sample Output
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:227]: Keep Heading-4

File: docs/searching-algorithms/exponential-search.mdx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+      search_value = int(input("Enter element to search\n"))
+
+      print(exponential_search(sorted(search_list), 0, len(search_list)-1, search_value))
+  main()
+  ```
+</TabItem>
+</Tabs>
+
+
+## Sample I/O
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:214]: Keep `Sample Input and Output`

File: docs/searching-algorithms/exponential-search.mdx [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+<TabItem value="cpp">
+
+  ```cpp
+  /*
+   Created by Sarthak-9
+   Exponential Search Algorithm is an optimized Binary Search to search an element in sorted array.
+   It is specifically used when the size of array is infinite.
+   Let n be the number of elements input in array and key be the element to be searched.
+   Time Complexity : O(log n)
+   Space Complexity : O(1)
```
Comment (HarshCasper) [inline docs/searching-algorithms/exponential-search.mdx:72]: No need to have this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/searching-algorithms/exponential-search.mdx (hunk 1 above)
Comment (ritvij14) [reply docs/searching-algorithms/exponential-search.mdx:41]: Done.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**